code/edgebridge/src/main/java/com/adobe/marketing/mobile/edge/bridge/EdgeBridgeExtension.java

```diff
@@ -166,7 +158,14 @@ void handleRulesEngineResponse(@NonNull final Event event) {
 	private void dispatchTrackRequest(final Map<String, Object> data, final long timestamp) {
 		Map<String, Object> xdmData = new HashMap<>();
 		xdmData.put("eventType", EdgeBridgeConstants.JsonValues.EVENT_TYPE);
-		xdmData.put("timestamp", iso8601DateFormat.format(new Date(timestamp)));
+		xdmData.put(
+			"timestamp",
+			TimeUtils.getIso8601Date(
+				new Date(timestamp),
+				"yyyy-MM-dd'T'HH:mm:ss.SSS'Z'",
+				TimeZone.getTimeZone("Etc/UTC")
+			)
```

Comment:
Can you use `TimeUtils.getIso8601DateTimeZoneISO8601()` here instead? It looks like that method returns the correct format and timezone automatically.

Reply:
Oh, I see, that method doesn't allow you pass in the `Date`.